### PR TITLE
YLP-2622 Remove references to `uploads` in the app

### PR DESCRIPTION
**Pull request by celia-rouis**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

File: packages/tideline/plugins/blip/basics/logic/datamunger.js [hunk 1]
```diff
@@ -31,8 +29,8 @@ function dataMunger(bgClasses, bgUnits = MGDL_UNITS) {
 
   return {
     getLatestPumpUploaded: function(patientData) {
-      const lastUploadDatum = getLatestPumpUpload(patientData.grouped.upload)
-      return _.get(lastUploadDatum, 'source', null)
+      const pumpSettings = patientData.grouped.pumpSettings[0]
```
Comment (TimCanu) [inline packages/tideline/plugins/blip/basics/logic/datamunger.js:32]: Are we always sure that `patientData.grouped.pumpSettings[0]` exists? Is there not a risk of an app crash if `patientData.grouped.pumpSettings` is null or empty?

File: packages/viz/src/modules/print/BasicsPrintView.js [hunk 2]
```diff
@@ -63,8 +62,8 @@ class BasicsPrintView extends PrintView {
       [SITE_CHANGE_TUBING]: Images.siteChangeTubingImage
     }
 
-    const latestPumpUpload = getLatestPumpUpload(_.get(data, 'data.upload.data', []))
-    this.source = _.get(latestPumpUpload, 'source', '').toLowerCase()
+    const pumpSettings = data.data.pumpSettings.data[0]
```
Comment (TimCanu) [inline packages/viz/src/modules/print/BasicsPrintView.js:65]: Same question than the previous one

File: packages/viz/src/utils/print/data.js [hunk 3]
```diff
@@ -189,7 +189,7 @@ export function selectDailyViewData(medicalData, startDate, endDate) {
     bgRange: processBgRange(dataByDate),
     bolusRange: processBolusRange(dataByDate),
     dateRange: [startDate.format('YYYY-MM-DD'), endDate.format('YYYY-MM-DD')],
-    latestPumpUpload: getLatestPumpUpload(medicalData.medicalData.uploads),
+    pumpSettings: medicalData.medicalData.pumpSettings[0],
```
Comment (TimCanu) [inline packages/viz/src/utils/print/data.js:192]: Same question

**Review by TimCanu — COMMENTED** [review]
Lot of code that I don't understand being deleted, very nice

File: packages/medical-domain/src/domains/repositories/medical/datum/device-parameter-change.service.ts [hunk 4]
```diff
@@ -79,9 +78,7 @@ const groupData = (data: DeviceParameterChange[]): DeviceParameterChange[] => {
   let currentGroup: DeviceParameterChange | null = null
   data.forEach((currentParam) => {
     const paramList = currentParam.params.map(p => {
-      if (p.id === currentParam.id) {
```
[the remark this answers is not in the document]
Comment (celia-rouis) [reply packages/medical-domain/src/domains/repositories/medical/datum/device-parameter-change.service.ts:82]: Yes, even though I must admit that I do not fully understand it:

When parsing the list of parameters, we are grouping the ones that occurred at the same time. What the code did before was that the timezone of the group was inferred from the timezone of one param only, but the timezone of the other params in the same group were not updated. I saw this bug while running the E2E tests on the SYSTREQ-86 ([86.C.3]) where, after removing the uploads, I had in the same parameter change group, one parameter displayed in a timezone and 2 parameters in another timezone. 
As to why it worked before removing the uploads, I do not have the answer at the moment (I don't remember it 😬) but I'll make sure to find it.

File: packages/tideline/plugins/blip/basics/logic/datamunger.js (hunk 1 above)
Comment (celia-rouis) [reply packages/tideline/plugins/blip/basics/logic/datamunger.js:32]: Given the implementation of the method `grouped()` (in `medical-data.service.ts`, line 126), as long as the `DatumType` "pumpSettings" exists, the value will at worst exist and be equal to an empty array. So worst case scenario, `patientData.grouped.pumpSettings[0]` will return `undefined`, but I don't think null pointer exception can happen here (please tell me if I'm mistaken!)

File: packages/viz/src/utils/print/data.js (hunk 3 above)
Comment (celia-rouis) [reply packages/viz/src/utils/print/data.js:192]: Given how `medicalData.medicalData` is initialized in `medical-data.service.ts` (line 81), I would say there is no risk of a null pointer exception: the `pumpSettings` property exists and is at worst equal to an empty array

File: packages/viz/src/modules/print/BasicsPrintView.js (hunk 2 above)
Comment (celia-rouis) [reply packages/viz/src/modules/print/BasicsPrintView.js:65]: In this case, the date received is from the `generateBasicsData()` method, which returns an object like this:
```ts
{
  data: {
    // ... other data
    pumpSettings: {
      data: medicalData.pumpSettings
    }
  }
}
```
And as I checked for your other comment, `medicalData.pumpSettings` is an empty array by default so I would say that there is no risk of a null pointer exception.

File: packages/tideline/plugins/blip/basics/logic/datamunger.js (hunk 1 above)
Comment (TimCanu) [reply packages/tideline/plugins/blip/basics/logic/datamunger.js:32]: Okay this works. I know that @ydrisR has made changes to this code too... To be kept in mind for his PR

**Review by TimCanu — APPROVED** [review]

**Pull request merged**